x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_event_log.ts

```diff
@@ -0,0 +1,620 @@
+
+    return {
+      countExecutionTimeouts: totalTimedoutExecutionsCount ?? 0,
+      countExecutionTimeoutsByType: parseSimpleRuleTypeBucket(aggregations.by_rule_type_id.buckets),
+    };
+  } catch (err) {
+    logger.warn(
+      `Error executing alerting telemetry task: getExecutionsTimeoutsPerDayCount - ${JSON.stringify(
+        err
+      )}`
+    );
+    return {
+      countExecutionTimeouts: 0,
+      countExecutionTimeoutsByType: {},
+    };
+  }
+}
+
+/**
+ * Bucket format:
+ * {
+ *   key: '.index-threshold',             // rule type id
+ *   doc_count: 78,                       // count of number of executions
+ *   avg_es_search_duration: {            // average es search duration across executions
+ *     value: 40.76056338028169,
+ *   },
+ *   percentile_alerts: {                 // stats for number of alerts created across executions
+ *     values: {
+ *       '50.0': 1,
+ *       '95.0': 1,
+ *       '99.0': 1,
+ *     },
+ *   },
+ *   execution_failures: {
+ *     doc_count: 7,                      // count of number of failed executions
+ *     by_reason: {
+ *       doc_count_error_upper_bound: 0,
+ *       sum_other_doc_count: 0,
+ *       buckets: [
+ *         {
+ *           key: 'execute',              // breakdown of reason for execution failures
+ *           doc_count: 4,
+ *         },
+ *         {
+ *           key: 'decrypt',
+ *           doc_count: 3,
+ *         },
+ *       ],
+ *     },
+ *   },
+ *   percentile_scheduled_actions: {      // stats for number of actions generated across executions
+ *     values: {
+ *       '50.0': 0,
+ *       '95.0': 0,
+ *       '99.0': 0,
+ *     },
+ *   },
+ *   avg_execution_time: {                // average execution time in nanoseconds across executions
+ *     value: 100576923.07692307,
+ *   },
+ *   avg_total_search_duration: {         // average total search duration across executions
+ *     value: 43.74647887323944,
+ *   },
+ * }
+ */
+
+export function parseRuleTypeBucket(
+  buckets: GetExecutionCountsAggregationBucket[]
+): Pick<
+  GetExecutionsPerDayCountResults,
+  | 'countRuleExecutionsByType'
+  | 'avgExecutionTimeByType'
+  | 'avgEsSearchDurationByType'
+  | 'avgTotalSearchDurationByType'
+  | 'generatedActionsPercentilesByType'
+  | 'alertsPercentilesByType'
+> {
+  return (buckets ?? []).reduce(
+    (summary, bucket) => {
+      const ruleType: string = replaceDotSymbols(bucket.key);
+      const numExecutions: number = bucket.doc_count ?? 0;
+      const avgExecutionTimeNanos = bucket?.avg_execution_time?.value ?? 0;
+      const avgEsSearchTimeMillis = bucket?.avg_es_search_duration?.value ?? 0;
+      const avgTotalSearchTimeMillis = bucket?.avg_total_search_duration?.value ?? 0;
+      const actionPercentiles = bucket?.percentile_scheduled_actions?.values ?? {};
+      const alertPercentiles = bucket?.percentile_alerts?.values ?? {};
+
+      return {
```

Comment:
This `reduce()` seems like it could just be a for loop; initialize the result with the reduce() initializer, and on each loop through, add the key to the relevant object property.

If the `reduce()` does still make sense, I think the `...summary` at the top probably isn't needed, as we're recalculating all the values each time through.  Seems like we're adding the existing result to a new object, then overwriting each property with a new version that includes the new type-specific value.